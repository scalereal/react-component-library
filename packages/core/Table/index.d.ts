import * as React from 'react';
import { WithStyle } from '../../utils';
import { Props } from './types';
declare const Table: React.FC<Props> & WithStyle;
export default Table;
