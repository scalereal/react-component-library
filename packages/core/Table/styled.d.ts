import { Props } from './types';
export declare const StyledThead: import("styled-components").StyledComponent<"thead", any, {}, never>;
export declare const StyledTh: import("styled-components").StyledComponent<"th", any, Props, never>;
export declare const StyledTr: import("styled-components").StyledComponent<"tr", any, Props, never>;
export declare const StyledTbody: import("styled-components").StyledComponent<"tbody", any, {}, never>;
export declare const StyledTd: import("styled-components").StyledComponent<"td", any, {}, never>;
export declare const StyledTable: import("styled-components").StyledComponent<"table", any, Props, never>;
