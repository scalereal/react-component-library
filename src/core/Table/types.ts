import { ReactNode } from "react";
import { HTMLProps, Omit, WithThemeProp } from '../../utils';


export type RadioSizes = 'XS' | 'S' | 'M' | 'L' | 'XL' ;
export type LabelPositions = 'left'| 'right'| 'top'| 'bottom';
export type displayStyle = 'inline' | 'block';

export interface Props extends Omit<HTMLProps<HTMLTableElement>,'size'|'onChange' >,WithThemeProp {
    children?:ReactNode;
     /** Options */
    thData?: {id:any,data:any}[];
    tdData?:{trId:any,trData:{tdId:any,tdData:any}[]}[];
    borderColor?:string
    thBgColor?:string;
    thTextColor?:string;
    hoverEffect?:boolean;
    nthTrBgColor?:string;
    hoverTrBgColor?:string;
}