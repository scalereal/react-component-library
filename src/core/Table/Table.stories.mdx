import Table from '.';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Core/Table"
    component={Table}
    argTypes={{
        hoverEffect: {
            control: { type: 'boolean' }
        },
        thBgColor: {
            control: 'color'
        },
        thTextColor: {
            control: 'color'
        },
        hoverTrBgColor: {
            control: 'color'
        },
        borderColor: {
            control: 'color'
        },
        nthTrBgColor: {
            control: 'color'
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        }
    }}
/>

# Table

Displays a Table.

export const Template = args => (
    <div>
        <Table {...args} />
    </div>
);

<Preview>
    <Story
        name="Table"
        args={{
            thData: [
                { id: 1, data: 'ID' },
                { id: 2, data: 'EmployeeID' },
                { id: 3, data: 'Name' }
            ],
            tdData: [
                {
                    trId: 1,
                    trData: [
                        { tdId: 1, tdData: '1' },
                        { tdId: 2, tdData: '2' },
                        { tdId: 3, tdData: '3' }
                    ]
                },
                {
                    trId: 2,
                    trData: [
                        { tdId: 1, tdData: '1' },
                        { tdId: 2, tdData: '2' },
                        { tdId: 3, tdData: '3' }
                    ]
                },
                {
                    trId: 3,
                    trData: [
                        { tdId: 1, tdData: '1' },
                        { tdId: 2, tdData: '2' },
                        { tdId: 3, tdData: '3' }
                    ]
                },
                {
                    trId: 4,
                    trData: [
                        { tdId: 1, tdData: '1' },
                        { tdId: 2, tdData: '2' },
                        { tdId: 3, tdData: '3' }
                    ]
                }
            ]
        }}
    >
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div>
        <Table
            thBgColor="green"
            thData={[
                { id: 1, data: 'ID' },
                { id: 2, data: 'EmployeeID' },
                { id: 3, data: 'Name' }
            ]}
            tdData={[
                {
                    trId: 1,
                    trData: [
                        { tdId: 1, tdData: '1' },
                        { tdId: 2, tdData: '2' },
                        { tdId: 3, tdData: '3' }
                    ]
                },
                {
                    trId: 2,
                    trData: [
                        { tdId: 1, tdData: '1' },
                        { tdId: 2, tdData: '2' },
                        { tdId: 3, tdData: '3' }
                    ]
                },
                {
                    trId: 3,
                    trData: [
                        { tdId: 1, tdData: '1' },
                        { tdId: 2, tdData: '2' },
                        { tdId: 3, tdData: '3' }
                    ]
                },
                {
                    trId: 4,
                    trData: [
                        { tdId: 1, tdData: '1' },
                        { tdId: 2, tdData: '2' },
                        { tdId: 3, tdData: '3' }
                    ]
                }
            ]}
        />
    </div>
</Preview>

### Props

<Props story="Table" />
