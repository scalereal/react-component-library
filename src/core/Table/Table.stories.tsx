import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';
import Table from '.';
import { Props } from './types';


export default {
    title: 'Components/Core/Table',
    component: Table,
    argTypes : {
        hoverEffect:{ 
            control: { type: 'boolean'}  
        },
        thBgColor : {
            control:'color'
        },
        thTextColor: {
            control:'color'
        },
        hoverTrBgColor: {
            control:'color'
        },
        borderColor:{
            control:'color'
        },
        nthTrBgColor:{
            control:'color'
        },
        ref:{
            table:{
              disable:true
            }
        },
        theme:{
            table:{
              disable:true
            }
        },
        
    }
} as Meta;

const Template: Story<Props> = ({...args }) => {
    // retrieves the appropriate icon passes it as a component prop
    return <Table {...args} />;
};

export const Default = Template.bind({});
Default.args={
    thData:[{id:1,data:'ID'},{id:2,data:'EmployeeID'},{id:3,data:'Name'}],
    tdData:[
        {trId:1,trData:[{tdId:1,tdData:'1'},{tdId:2,tdData:'2'},{tdId:3,tdData:'3'}]},
        {trId:2,trData:[{tdId:1,tdData:'1'},{tdId:2,tdData:'2'},{tdId:3,tdData:'3'}]},
        {trId:3,trData:[{tdId:1,tdData:'1'},{tdId:2,tdData:'2'},{tdId:3,tdData:'3'}]},
        {trId:4,trData:[{tdId:1,tdData:'1'},{tdId:2,tdData:'2'},{tdId:3,tdData:'3'}]},
    ],
    hoverEffect:true
}


