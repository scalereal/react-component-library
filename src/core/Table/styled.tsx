import styled from 'styled-components';
import { defaultTheme } from '../../theme/theme';
import { Props } from './types';

export const StyledThead = styled.thead`

`;

export const StyledTh = styled.th<Props>`
    padding-top: 12px;
    padding-bottom: 12px;
    text-align: left;
    background-color: ${pr => pr.thBgColor || defaultTheme.primaryColor};
    color: ${pr => pr.thTextColor || defaultTheme.whiteColor};
`;

export const StyledTr = styled.tr<Props>`
    &:nth-child(even){background-color: #f2f2f2;};
    /* &:hover {background-color:#ddd;} */
    ${pr => pr.hoverEffect?`&:hover {background-color:${pr.hoverTrBgColor || '#ddd'};`:''}
`;

export const StyledTbody = styled.tbody`

`;

export const StyledTd = styled.td`

`;

export const StyledTable = styled.table<Props>`
    font-family: Arial, Helvetica, sans-serif;
    border-collapse: collapse;
    width: 100%;
    ${StyledTd},${StyledTh}{
        border: 1px solid ${pr => pr.borderColor || '#ddd'};
        padding: 8px;
    }
`;