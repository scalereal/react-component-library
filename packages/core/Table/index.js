var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from 'react';
import { StyledTable, StyledTbody, StyledTd, StyledTh, StyledThead, StyledTr } from './styled';
var Table = React.memo(React.forwardRef(function (props, ref) {
    var thData = props.thData, tdData = props.tdData;
    return (React.createElement(StyledTable, __assign({ ref: ref }, props),
        React.createElement(StyledThead, null,
            React.createElement(StyledTr, __assign({}, props), thData && thData.map(function (th) { return React.createElement(StyledTh, __assign({ key: th.id }, props), th.data); }))),
        React.createElement(StyledTbody, null, tdData && tdData.map(function (tr) { return React.createElement(StyledTr, __assign({ key: tr.trId }, props),
            " ",
            tr.trData.map(function (td) { return React.createElement(StyledTd, __assign({ key: td.tdId }, props), td.tdData); }),
            " "); }))));
}));
Table.displayName = "Table";
Table.defaultProps = {
// hoverEffect: false
};
export default Table;
