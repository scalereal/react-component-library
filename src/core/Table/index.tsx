import * as React from 'react';
import { WithStyle } from '../../utils';
import { StyledTable, StyledTbody, StyledTd, StyledTh, StyledThead, StyledTr } from './styled';
import { Props } from './types';

export const Table:React.FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {thData,tdData} = props;
    return <StyledTable ref={ref} {...props}>
                <StyledThead>
                    <StyledTr {...props}>
                        {thData && thData.map(th=><StyledTh key={th.id} {...props}>{th.data}</StyledTh>)}
                    </StyledTr>
                </StyledThead>
                <StyledTbody>
                    {tdData && tdData.map(tr=><StyledTr key={tr.trId} {...props}>{tr.trData.map(td => <StyledTd key={td.tdId} {...props} >{td.tdData}</StyledTd>)}</StyledTr>)}
                </StyledTbody>
            </StyledTable>
}));

Table.displayName="Table";
Table.defaultProps = {
    hoverEffect: true
};

export default Table