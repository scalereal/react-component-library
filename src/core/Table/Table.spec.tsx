import { render } from '@testing-library/react';
import * as React from 'react';
import Table from '.';

describe('CheckBoxGroup component', () => {
    it('should render with default theme', () => {
        const { container } = render(<Table />);
        expect(container).toMatchSnapshot();
    });

});