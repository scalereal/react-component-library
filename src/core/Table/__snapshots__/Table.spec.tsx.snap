// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxGroup component should render with all the props given 1`] = `
<div>
  <table
    class="sc-jSgupP eKkiRr"
  >
    <thead
      class="sc-bdfBwQ dbXZak"
    >
      <tr
        class="sc-dlfnbm fgYrRv"
      >
        <th
          class="sc-gsTCUz dPAonl"
        >
          ID
        </th>
        <th
          class="sc-gsTCUz dPAonl"
        >
          EmployeeID
        </th>
        <th
          class="sc-gsTCUz dPAonl"
        >
          Name
        </th>
      </tr>
    </thead>
    <tbody
      class="sc-hKgILt cTJgxF"
    >
      <tr
        class="sc-dlfnbm fgYrRv"
      >
         
        <td
          class="sc-eCssSg flCDJa"
        >
          1
        </td>
        <td
          class="sc-eCssSg flCDJa"
        >
          2
        </td>
        <td
          class="sc-eCssSg flCDJa"
        >
          3
        </td>
         
      </tr>
      <tr
        class="sc-dlfnbm fgYrRv"
      >
         
        <td
          class="sc-eCssSg flCDJa"
        >
          1
        </td>
        <td
          class="sc-eCssSg flCDJa"
        >
          2
        </td>
        <td
          class="sc-eCssSg flCDJa"
        >
          3
        </td>
         
      </tr>
      <tr
        class="sc-dlfnbm fgYrRv"
      >
         
        <td
          class="sc-eCssSg flCDJa"
        >
          1
        </td>
        <td
          class="sc-eCssSg flCDJa"
        >
          2
        </td>
        <td
          class="sc-eCssSg flCDJa"
        >
          3
        </td>
         
      </tr>
      <tr
        class="sc-dlfnbm fgYrRv"
      >
         
        <td
          class="sc-eCssSg flCDJa"
        >
          1
        </td>
        <td
          class="sc-eCssSg flCDJa"
        >
          2
        </td>
        <td
          class="sc-eCssSg flCDJa"
        >
          3
        </td>
         
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CheckBoxGroup component should render with default theme 1`] = `
<div>
  <table
    class="sc-jSgupP fPFcvH"
  >
    <thead
      class="sc-bdfBwQ dbXZak"
    >
      <tr
        class="sc-dlfnbm eeeREN"
      />
    </thead>
    <tbody
      class="sc-hKgILt cTJgxF"
    />
  </table>
</div>
`;
